packages/content-conformance/src/rules/__tests__/ensure-valid-link-format.test.ts

```diff
@@ -0,0 +1,157 @@
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid onboarding tutorial link](/onboarding/foo/bar)`,
+        messages: [],
+      },
+      {
+        fixture: `[I am a valid link to an .io domain](https://waypointproject.io/)
+        [I am a valid link to another .io domain](https://terraform.io)`,
+        messages: [],
+      },
+    ])
+  })
+
+  test('errors for learn.hashicorp.com links', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[learn link](https://learn.hashicorp.com/tutorials/waypoint/aws-ecs)`,
+        messages: [/Unexpected link to `learn\.hashicorp\.com`/],
+      },
+      {
+        fixture: `[invalid learn link](https://learn.hashicorp.com/tutorials/waypoint)`,
+        messages: [/Unexpected link to `learn\.hashicorp\.com`/],
+      },
+    ])
+  })
+
+  test('errors for .io site docs links', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[.io docs link](https://www.waypointproject.io/docs)`,
+        messages: [/Unexpected link to documentation on `waypointproject\.io`/],
+      },
+      {
+        fixture: `[.io docs link](https://waypointproject.io/docs)`,
+        messages: [/Unexpected link to documentation on `waypointproject\.io`/],
+      },
+      {
+        fixture: `[Another .io docs link](https://www.consul.io/commands)`,
+        messages: [/Unexpected link to documentation on `consul\.io`/],
+      },
+    ])
+  })
+
+  test('errors for fully qualified developer.hashicorp.com links', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[full developer link](https://developer.hashicorp.com)`,
+        messages: [
+          /Unexpected fully-qualified link to `developer\.hashicorp\.com`/,
+        ],
+      },
+      {
+        fixture: `[another full developer link](https://developer.hashicorp.com/tutorials/library)`,
+        messages: [
+          /Unexpected fully-qualified link to `developer\.hashicorp\.com`/,
+        ],
+      },
+      {
+        fixture: `[full developer link](https://developer.hashicorp.com/waypoint/docs)`,
+        messages: [
+          /Unexpected fully-qualified link to `developer\.hashicorp\.com`/,
+        ],
+      },
+    ])
+  })
+
+  test('errors for links without product base paths', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[relative docs link](/docs/foo)`,
+        messages: [/Unexpected product-relative link/],
+      },
+      {
+        fixture: `[relative docs link](/api-docs)`,
+        messages: [/Unexpected product-relative link/],
+      },
+      {
+        fixture: `[relative docs link](/commands)`,
+        messages: [/Unexpected product-relative link/],
+      },
+    ])
+  })
+
+  test('errors for folder-relative links', () => {
+    testRule(ensureValidLinkFormat, [
+      {
+        fixture: `[folder-relative link](./foo/bar)`,
+        messages: [/Unexpected folder-relative link/],
+      },
```

Comment:
🚧 Blocking: I added a couple of cases here locally, and see that there's a failure after doing so. Looks like it's the no-dots case that's failing.

```suggestion
      {
        fixture: `[dot-slash folder-relative link](./foo/bar)`,
        messages: [/Unexpected folder-relative link/],
      },
      {
        fixture: `[dot-dot-slash folder-relative link](../foo/bar)`,
        messages: [/Unexpected folder-relative link/],
      },
      {
        fixture: `[no-dots folder-relative link](foo/bar)`,
        messages: [/Unexpected folder-relative link/],
      },
```

Reply:
Great catch! 👀